gluten-ut/spark35/src/test/scala/org/apache/spark/sql/execution/datasources/GlutenNoopWriterRuleSuite.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.execution.datasources
+
+import org.apache.spark.sql.{GlutenSQLTestsBaseTrait, SaveMode}
+import org.apache.spark.sql.execution.QueryExecution
+import org.apache.spark.sql.util.QueryExecutionListener
+
+class GlutenNoopWriterRuleSuite extends GlutenSQLTestsBaseTrait {
+
+  class WriterColumnarListener extends QueryExecutionListener {
+    var fakeRowAdaptor: Option[FakeRowAdaptor] = None
+
+    override def onSuccess(funcName: String, qe: QueryExecution, durationNs: Long): Unit = {
+      fakeRowAdaptor = qe.executedPlan.collectFirst { case f: FakeRowAdaptor => f }
```

Comment:
@jackylee-ch FakeRowAdaptor is used in spark 32 and 33. Why we need to add this check in spark 35 test folder?

Reply:
The `GlutenNoopWriterRule` would add a `FakeRowAdaptor` after v2 write command while writing to noop source.